autoload/targets.vim

```diff
@@ -39,16 +62,25 @@ function! targets#e(mapmode, modifier, original)
         return a:original
     endif
 
-    let char1 = nr2char(getchar())
-    let [trigger, which, chars] = [char1, 'c', char1]
-    for i in range(2)
-        if g:targets_nl[i] ==# trigger
-            " trigger was which, get another char for trigger
-            let char2 = nr2char(getchar())
-            let [trigger, which, chars] = [char2, 'nl'[i], chars . char2]
-            break
-        endif
-    endfor
+    let [nKeys, lKeys] = targets#getKeysAsList(g:targets_nl)
+    let keys = [s:getKeyAsStr()]
+    let chars = join(keys, '')
+    while (len(chars) <= len(nKeys) && s:hasPrefix(nKeys, chars)) || 
+          \ (len(chars) <= len(lKeys) && s:hasPrefix(lKeys, chars))
+        let nextKey = s:getKeyAsStr()
+        let keys = keys + [nextKey]
+        let chars = chars . nextKey
+    endwhile
+    if s:hasPrefix(chars, nKeys) 
+      let which = 'n'
+      let trigger = keys[len(keys)-1]
+    elseif s:hasPrefix(chars, lKeys) 
+      let which = 'l'
+      let trigger = keys[len(keys)-1]
+    else
+      let which = 'c'
+      let trigger = keys[0]
```

Comment:
Since currently trigger must be a single char, I think we might want to add another `elseif len(keys) > 1` where we `return typed` (and move the `typed` definition above this `if` block). So if I have `g:targets_nl=[',n', ',l']` and type `vi,x` then it should bail out instead of trying `vi,` (if I have `,` as prefix of my `nl` values then `,` can't be a trigger anymore).

Edit: This is implemented in the suggested `while 1` loop.